hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/Exceptions.kt

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.indexer.services
+
+import org.springframework.http.HttpStatus.CONFLICT
+import org.springframework.http.HttpStatus.NOT_FOUND
+import org.springframework.web.server.ResponseStatusException
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.IndexStatus
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.SyncIndex
+
+class BuildAlreadyInProgressException(indexStatus: IndexStatus) :
+  ResponseStatusException(CONFLICT, "The build for ${indexStatus.otherIndex} is already ${indexStatus.otherIndexState} (started at ${indexStatus.otherIndexStartBuildTime})")
+
+class BuildInProgressException(indexStatus: IndexStatus) :
+  ResponseStatusException(CONFLICT, "The build for ${indexStatus.otherIndex} is already ${indexStatus.otherIndexState} (started at ${indexStatus.otherIndexStartBuildTime})")
+
+class BuildCancelledException(indexStatus: IndexStatus) :
+  ResponseStatusException(CONFLICT, "The build for ${indexStatus.otherIndex} is in state ${indexStatus.otherIndexState} (started at ${indexStatus.otherIndexStartBuildTime})")
+
+class BuildAbsentException(indexStatus: IndexStatus) :
+  ResponseStatusException(CONFLICT, "The build for ${indexStatus.otherIndex} is in state ${indexStatus.otherIndexState} (started at ${indexStatus.otherIndexStartBuildTime})")
```

Comment:
Maybe push this message up into `IndexStatus` to avoid the duplication here?

Reply:
yes, good idea